.github/workflows/ci.yaml

```diff
@@ -177,6 +177,28 @@ jobs:
           name: npm-package
           path: ./package.tar.gz
 
+  npm:
+    # the npm-package gets uploaded as an artifact in Build
+    # so we need that to complete before this runs
+    needs: build
+    # This environment "npm" requires someone from
+    # coder/code-server-reviewers to approve the PR before this job runs.
+    environment: npm
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Run ./ci/steps/publish-npm.sh
+        run: yarn publish:npm
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          # The way this logic works is it checks if the GitHub event is a push to the `main` branch
+          # if so, we're in the staging environment (i.e. running in CI on merge into `main`)
+          # otherwise it's running in CI on a PR event, meaning development environment
+          # Source: https://kevsoft.net/2020/06/10/running-github-action-steps-and-jobs-only-on-push-to-master.html
+          ENVIRONMENT: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' && 'staging' || 'development' }}
+
```

Comment:
We now have one job instead of the two workflows in my first implementation (see deletion of `npm-beta.yaml` and `npm-dev.yaml`). This allows us to wait to run this step until have the `npm-package` artifact has been uploaded in the `build` job.